src/math/Vector3.js

```diff
@@ -5,87 +5,264 @@
  * @author mikael emtinger / http://gomo.se/
  * @author egraether / http://egraether.com/
  * @author WestLangley / http://github.com/WestLangley
+ * @author Ben Houston / https://clara.io
  */
 
-THREE.Vector3 = function ( x, y, z ) {
+ THREE.Vector3 = function ( array, x, y, z ) {
 
-	this.x = x || 0;
-	this.y = y || 0;
-	this.z = z || 0;
+	 if( !( ( array instanceof Float32Array ) || ( array === undefined ) ) ) {
+
+		 z = y; y = x; x = array; array = null;
+
+	 }
+
+	this.array = array || THREE.DefaultAllocator.getFloat32( 3 );
+
+	 if( x !== undefined ) this.set( x, y, z );
+
+	 return this;
+
+ };
```

Comment:
Initialized to zero by default: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Float32Array  To prevent data leaks I guess between sessions.